core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -161,6 +190,8 @@ public void close() throws IOException {
     mFileSystemMasterClientPool = null;
     mBlockMasterClientPool.close();
     mBlockMasterClientPool = null;
+    mMetricsMasterClient.close();
+    mMetricsMasterClient = null;
```

Comment:
What do we want to achieve here, since `close()` will probably cause the heartbeat thread to fail, should we interrupt the heartbeat thread first?